clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/OpenstackServerGroupNameResolver.groovy

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 Veritas Technologies LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.helpers.AbstractServerGroupNameResolver
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackCredentials
+import org.openstack4j.model.heat.Stack
+
+import java.text.SimpleDateFormat
+
+class OpenstackServerGroupNameResolver extends AbstractServerGroupNameResolver {
+  private static final String PHASE = "DEPLOY"
+
+  private final String region
+  private final OpenstackCredentials credentials
+
+  OpenstackServerGroupNameResolver(OpenstackCredentials credentials, String region) {
+    this.credentials = credentials
+    this.region = region
+  }
+
+  @Override
+  String getPhase() {
+    return PHASE
+  }
+
+  @Override
+  String getRegion() {
+    return region
```

Comment:
I think this stackoverflow due to Groovy turning `region` into `getRegion`? I have run into that elsewhere, but still needing the getter due to the abstract method. You can get around it with `this.@region` I think, which forces Groovy to use the field, not the getter.